drivers/include/isl29125.h

```diff
@@ -0,0 +1,165 @@
+ *
+ * @author      Ludwig Ortmann <ludwig.ortmann@fu-berlin.de
+ */
+
+#ifndef ISL29125_H
+#define ISL29125_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "color.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+ /**
+  * @brief The sensors hard coded I2C address
+  */
+#define ISL29125_I2C_ADDRESS 0x44
+
+/**
+ * @brief data type for storing lux rgb sensor readings
+ */
+typedef struct {
+    float red;              /**< red lux value */
+    float green;            /**< green lux value */
+    float blue;             /**< blue lux value */
+} isl29125_rgb_t;
+
+/**
+ * @brief supported operation modes of the ISL29125 sensor's AD
+ * conversion
+ */
+typedef enum {
+    ISL29125_MODE_DOWN = 0x00,      /**< ADC powered down */
+    ISL29125_MODE_STANDBY = 0x04,   /**< AD conversion not performed */
+    ISL29125_MODE_RGB = 0x05,       /**< RGB conversion */
+    ISL29125_MODE_R = 0x02,         /**< red conversion only */
+    ISL29125_MODE_G = 0x01,         /**< green conversion only */
+    ISL29125_MODE_B = 0x03,         /**< blue conversion only */
+    ISL29125_MODE_RG = 0x06,        /**< red and green conversion only */
+    ISL29125_MODE_GB = 0x07         /**< green and blue conversion only */
+} isl29125_mode_t;
+
+/**
+ * @brief supported RGB sensing range values of the ISL29125 sensor
+ */
+typedef enum {
+    ISL29125_RANGE_375 = 0x00,      /**< range: 5.7m - 375 lux */
+    ISL29125_RANGE_10K = 0x08       /**< range: 0.152 - 10,000 lux */
+} isl29125_range_t;
+
+/**
+ * @brief supported color resolutions of the ISL29125 sensor's AD
+ * conversion
+ */
+typedef enum {
+    ISL29125_RESOLUTION_12 = 0x10,  /**< resolution: 12 bit */
+    ISL29125_RESOLUTION_16 = 0x00   /**< resolution: 16 bit */
+} isl29125_resolution_t;
+
+/**
+ * @brief Device descriptor for ISL29125 sensors
+ */
+typedef struct {
+    i2c_t i2c;                      /**< I2C device the sensor is connected to */
+    uint8_t address;                /**< I2C bus address of the sensor */
+    gpio_t gpio;                    /**< GPIO pin for interrupt/sync mode */
+    isl29125_range_t range;         /**< sensor range */
+    isl29125_resolution_t res;      /**< sensor resolution */
+} isl29125_t;
+
+/**
+ * @brief initialize a new ISL29125 device
+ *
+ * @param[in] dev           device descriptor of an ISL29125 device
+ * @param[in] i2c           I2C device the sensor is connected to
+ * @param[in] address       I2C address of the sensor
+ * @param[in] gpio          GPIO pin for interrupt/sync mode (currently unused)
+ * @param[in] mode          operation mode
+ * @param[in] range         measurement range
+ * @param[in] resolution    ADC conversion resolution
+ *
+ * @return              0 on success
+ * @return              -1 on error
+ */
+int isl29125_init(isl29125_t *dev, i2c_t i2c, uint8_t address, gpio_t gpio,
+        isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution);
```

Comment:
At least it's not in the [Coding Conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions).